pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -224,6 +217,30 @@ func (pv *Provisioner) cloneVirtualMachine(s *SessionContext, cluster *clusterv1
 		return fmt.Errorf("error fetching virtual machine or template properties: %s", err)
 	}
 
+	host, err := src.HostSystem(ctx)
+	if err != nil {
+		return fmt.Errorf("HostSystem failed. err=%s", err)
+	}
+
+	pool, err := s.finder.ResourcePoolOrDefault(ctx, machineConfig.MachineSpec.ResourcePool)
+	if err != nil {
+		klog.Warningf("Failed to find ResourcePool=%s err=%s. Attempting to create it.", machineConfig.MachineSpec.ResourcePool, err)
+
+		poolRoot, errRoot := host.ResourcePool(ctx)
+		if errRoot != nil {
+			klog.Errorf("Failed to find root ResourcePool. err=%s", errRoot)
+			return errRoot
+		}
+
+		pool, err = poolRoot.Create(ctx, machineConfig.MachineSpec.ResourcePool, types.DefaultResourceConfigSpec())
```

Comment:
I would also log the fact that we're using default values for the resource pool.  It feels like we should expand the spec to allow users to specify the cpu and memory allocation value of the pool in the yaml files.  I'll add a new issue for this and link it to this issue.

Reply:
Yup sounds good!

Reply:
updated